src/Bridge/LibraryBridgeHelper.cpp

```diff
@@ -0,0 +1,199 @@
+    return uri;
+}
+
+
+void LibraryBridgeHelper::startBridge(std::unique_ptr<ShellCommand> cmd) const
+{
+    context.addBridgeCommand(std::move(cmd));
+}
+
+
+bool LibraryBridgeHelper::initLibrary(const std::string & library_path, const std::string library_settings)
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", LIB_NEW_METHOD);
+    uri.addQueryParameter("library_path", library_path);
+    uri.addQueryParameter("library_settings", library_settings);
+
+    return executeRequest(uri);
+}
+
+
+bool LibraryBridgeHelper::cloneLibrary(const std::string & other_dictionary_id)
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", LIB_CLONE_METHOD);
+    uri.addQueryParameter("from_dictionary_id", other_dictionary_id);
+
+    return executeRequest(uri);
+}
+
+
+bool LibraryBridgeHelper::removeLibrary()
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", LIB_DELETE_METHOD);
+
+    return executeRequest(uri);
+}
+
+
+bool LibraryBridgeHelper::isModified()
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", IS_MODIFIED_METHOD);
+
+    return executeRequest(uri);
+}
+
+
+bool LibraryBridgeHelper::supportsSelectiveLoad()
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", SUPPORTS_SELECTIVE_LOAD_METHOD);
+
+    return executeRequest(uri);
+}
+
+
+BlockInputStreamPtr LibraryBridgeHelper::loadAll(const std::string attributes_string, const Block & sample_block)
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+
+    uri.addQueryParameter("method", LOAD_ALL_METHOD);
+    uri.addQueryParameter("attributes", attributes_string);
+    uri.addQueryParameter("sample_block", sample_block.getNamesAndTypesList().toString());
+
+    return loadBase(uri, sample_block);
+}
+
+
+BlockInputStreamPtr LibraryBridgeHelper::loadIds(const std::string attributes_string, const std::string ids_string, const Block & sample_block)
+{
+    startBridgeSync();
+
+    auto uri = getDictionaryURI();
+
+    uri.addQueryParameter("method", LOAD_IDS_METHOD);
+    uri.addQueryParameter("attributes", attributes_string);
+    uri.addQueryParameter("ids", ids_string);
+    uri.addQueryParameter("sample_block", sample_block.getNamesAndTypesList().toString());
+
+    return loadBase(uri, sample_block);
+}
+
+
+BlockInputStreamPtr LibraryBridgeHelper::loadKeys(const Block & key_columns, const Block & sample_block)
+{
+    startBridgeSync();
+
+    auto columns = key_columns.getColumns();
+    auto keys_sample_block = key_columns.cloneEmpty();
+
+    auto uri = getDictionaryURI();
+    uri.addQueryParameter("method", LOAD_KEYS_METHOD);
+    uri.addQueryParameter("sample_block", sample_block.getNamesAndTypesList().toString());
+    uri.addQueryParameter("requested_block", keys_sample_block.getNamesAndTypesList().toString());
```

Comment:
Seems like we can directly call `getNamesAndTypeList` from `key_columns` why we need to clone it?